### PR TITLE
Bun.cron.parse: return null at the upper Date boundary with { tz }

### PR DESCRIPTION
**Pull request by robobun**
## What

`Bun.cron.parse()` with a `{ tz }` option crashed a debug assert (and returned `Invalid Date` in release) when `from` was at or near the upper ECMAScript Date boundary and the next match fell past it. The local-time path already returned `null` there; the `{ tz }` path introduced in #35122 did not.

## Repro

```sh
$ bun -e 'console.log(Bun.cron.parse("* * * * *", 8.64e15, { tz: "UTC" }))'
```

Debug:
```
ASSERTION FAILED: isYearWithinLimits(year) || year == outOfRangeYear
JavaScriptCore/ISO8601.h(370) : JSC::ISO8601::PlainDate::PlainDate(int32_t, unsigned int, unsigned int)
```

Release: `Invalid Date` (expected `null`, same as `TZ=UTC bun -e 'console.log(Bun.cron.parse("* * * * *", 8.64e15))'`).

Also reproduces with `America/New_York`, `Asia/Tokyo`, and with schedules whose next match is in year 275761 (e.g. `"0 0 1 1 *"` at `8.64e15`).

## Cause

`CronExpression::next()` walks wall-clock time forward from `from_ms`. At `from_ms = 8.64e15` (which is `+275760-09-13T00:00:00Z`), every candidate past that instant is out of the ECMAScript Date range. For a named tz, `Bun__gregorianDateTimeToMSInZone` calls `getEpochNanosecondsFor`, which returns a range error that the binding maps to NaN. `resolve_local_match` compared that NaN with `> from_ms` (false) and reported `None`, which `next()` interprets as "DST-skipped minute, keep walking". The walk then advanced until the year rolled past 275760, at which point the binding constructs `JSC::ISO8601::PlainDate { 275761, ... }` whose constructor asserts `isYearWithinLimits`.

## Fix

- `Bun__gregorianDateTimeToMSInZone`: return NaN for years outside `JSC::ISO8601::isYearWithinLimits` instead of constructing a `PlainDate` that asserts; `getEpochNanosecondsFor` would have produced a range error (NaN) for that date anyway.
- `resolve_local_match`: surface NaN as `Some(NaN)` so `next()` sees "out of range" rather than "keep walking".
- `next()`: clamp the returned instant to `MAX_ECMASCRIPT_TIME` (`None` past it, which also catches NaN). Both callers now see `None` for an out-of-range match; the per-caller clamp in `cron_parse` is subsumed and removed.

Also adds an ICU warm-up before the "Feb 30 returns null quickly" timing assertion; that test's 50 ms bound was being spent on ICU's first-open under debug+ASAN.

## Verification

`bun bd test test/js/bun/cron/cron-parse.test.ts` (and `cron-local-time.test.ts`, `cron.test.ts`): 26 / 84 / 63-skipped pass. The new `{ tz }` boundary cases crash with the PlainDate assertion on main.

### CODE REVIEW

File: src/jsc/bindings/bindings.cpp [hunk 1]
```diff
@@ -5824,6 +5824,10 @@ extern "C" [[ZIG_EXPORT(nothrow)]] double Bun__gregorianDateTimeToMSInZone(JSC::
     int year, int month, int day, int hour, int minute, int second, int millisecond, uint32_t tzID)
 {
     UNUSED_PARAM(globalObject);
+    // PlainDate's constructor asserts the year; getEpochNanosecondsFor would
+    // return a range error anyway, so report out-of-range as NaN up front.
```
[the remark this answers is not in the document]
Comment (robobun) [reply src/jsc/bindings/bindings.cpp]: Dropped in f5125aa; the `isYearWithinLimits` check plus the `PlainDate` construction on the next line carry the context.

File: src/runtime/api/cron_parser.rs [hunk 2]
```diff
@@ -219,7 +220,14 @@ impl CronExpression {
                 }
             }
         }
-        Ok(if result > from_ms { Some(result) } else { None })
+        // NaN here means `dt` maps past the ECMAScript Date range (the Named
+        // tz path returns NaN for out-of-range epochs). Surface it so `next()`
+        // stops walking instead of treating it like a DST-skipped minute.
```
[the remark this answers is not in the document]
Comment (robobun) [reply src/runtime/api/cron_parser.rs]: Trimmed to one line in f5125aa. The `is_nan()` disjunct is load-bearing (None means keep walking, NaN means stop), so a note stays.

**Comment by robobun** [comment]
Heads up: #40402 adds gap handling to `resolve_local_match` above the tail return this PR changes. A NaN `result` bypasses the new early returns (`CronTz::in_gap` returns false for NaN), so the `result.is_nan() || result > from_ms` tail still decides for it. The hunks do not overlap. Both PRs should merge in either order without a change.

**Comment by robobun** [comment]
#42297 changes the same function and overlaps with this PR.

- #42297 stops the day walk in `CronExpression::next()` at +275760-09-13. The walk then cannot reach year 275761, so the assertion in this PR's description cannot fire from `next()`. On `main` today the walk trips an earlier assertion, `year >= minYear && year <= maxYear` at `PlainGregorianDateTime.h(91)`, inside the `ms_to_gregorian_date_time_utc` call. #42297 fixes that one too.
- The `{ tz }` boundary rows from this PR's test are part of the new test in #42297. They pass there.
- This PR does not change the day walk. So it does not stop the endless loop that the WebKit upgrades (#42177, #42267) cause in `next()`.
- One change stays unique to this PR: the move of the `> MAX_ECMASCRIPT_TIME` clamp from `cron_parse` into `next()`.

If #42297 merges, rebase this PR down to that one change, or close it.

**Comment by robobun** [comment]
Closing in favor of #42297. It bounds the day walk in `CronExpression::next()` and includes the `{ tz }` boundary cases from this PR.

Checked on `main` @ 158ff6cdef (WebKit cf1b36ec8703, debug build):

- #42297 rebased on `main`: `test/js/bun/cron/` passes (136 pass, 63 skipped). The skipped tests need `crontab`, `launchctl` or `schtasks`. Every repro from this PR's description returns `null`: `* * * * *` and `0 0 1 1 *` from 8.64e15, in local time and with `UTC`, `America/New_York` and `Asia/Tokyo`.
- This PR's diff alone on `main`: `Bun.cron.parse("0 0 1 1 *", 8.64e15)` and `Bun.cron.parse("0 0 30 2 *", new Date(8.64e15 - 86400000 * 400))` do not return (killed after 60 s), with and without `{ tz }`. Since #42319, JSC returns year 0 for a time value more than one day past the Date range, and this PR does not stop the day walk.
- The four rows of this PR's test are in the new test of #42297. Its zone list has `Pacific/Kiritimati` (UTC+14) and not `Asia/Tokyo` (UTC+9). I ran all rows of both tests with `Asia/Tokyo` on the #42297 build, and they pass.
- The year guard in `Bun__gregorianDateTimeToMSInZone` is not needed. `CronExpression::next()` is the only caller, and with #42297 it passes no year after 275760. On the current WebKit the `PlainDate` constructor maps an out-of-range year to a sentinel and has no assertion.
